xla/service/spmd/dot_handler.cc

```diff
@@ -1909,22 +1909,56 @@ absl::StatusOr<HloInstruction*> PartitionBaseCase(
     }
   }
 
+  // If we see a dot that shares the same operand with a windowed einsum ag loop
+  // and disable_ag_rewrite_for_multiple_consumers is true. We skip rewriting
+  // the current dot.
+  bool should_skip_windowed_einsum = false;
+  if (options.disable_ag_rewrite_for_multiple_consumers) {
+    auto lhs_operand = lhs.hlo()->opcode() == HloOpcode::kReshape ||
```

Comment:
Why are these instructions skipped? Please add a comment explaining why.

Reply:
added a comment for this logic. We skip those since they are optional and dont change the functional meaning of the ag-dot pattern, essentially we can support ag-(reshape)-dot.